data/src/main/java/io/homeassistant/companion/android/data/themes/ThemesRepositoryImpl.kt

```diff
@@ -0,0 +1,30 @@
+package io.homeassistant.companion.android.data.themes
+
+import io.homeassistant.companion.android.data.LocalStorage
+import io.homeassistant.companion.android.domain.themes.ThemesRepository
+import javax.inject.Inject
+import javax.inject.Named
+
+class ThemesRepositoryImpl @Inject constructor(
+    @Named("themes") private val localStorage: LocalStorage,
+    @Named("osVersion") private val osVersion: String
+) : ThemesRepository {
+
+    companion object {
+        private const val PREF_THEME = "theme"
+        private const val P = 28
+    }
+
+    override suspend fun getCurrentTheme(): String {
+        val theme = localStorage.getString(PREF_THEME)
+        return if (theme.isNullOrEmpty()) {
+            if (osVersion.toInt() >= P) {
+                "system"
+            } else "light"
+        } else theme
+    }
```

Comment:
Now that we are adding a version check into the default I think we would want the default to come from `SettingsPresenterImpl.kt`.  Either return `default` when unset or make the return nullable then deal with it in the presenter.

Reply:
`getCurrentTheme()` is also called in the `WebViewPresenterImpl`. If i implement the check on `SettingsPresenterImpl`, i also need to implement it in `WebViewPresenterImpl`. Any suggestion how to implement this only one time? Maybe in the ThemesUseCase?

Reply:
Put the logic in ThemeHandler maybe?